### PR TITLE
adding a gremlin REST option. Sending better error messages. npm audit

### PR DESCRIPTION
**Pull request by thiagomata**
In the last version of the gremlin server, there are some problems using the WebSocket version. So, this branch uses the REST API to provide a similar experience. Also, it provides better data handling with the error messages when the query fails and add some logs to help to see what is happening.

The current version was tested with Gremlin 3.2.5 using the docker https://hub.docker.com/r/bricaud/gremlin-server-with-demo-graph/.
image

### CODE REVIEW

**Comment by cscetbon** [comment]
@thiagomata you sure those changes work with AWS Neptune ? Other than that, I think those changes are great

**Comment by cscetbon** [comment]
I also don't see a way to enable debug logs in the current version, but yours seems to add that feature too. Can you confirm it ?

**Comment by thiagomata** [comment]
Yes. I added some very basic logging when getting error messages.

**Comment by thiagomata** [comment]
> @thiagomata you sure those changes work with AWS Neptune ? Other than that, I think those changes are great

Yes, I was using for that.

**Comment by triggan** [comment]
One change that needs to be made before this PR is committed - Neptune specifically defaults to TLS/SSL only connectivity as of Feb. of 2020.  Moreover, as of Neptune version 1.0.4.0 released in Oct. 2020, non-TLS traffic is no longer permitted.  As such, the use of ws:// needs to be changed to wss:// or there needs to be some sort of flag to set the use of TLS or non-TLS for other TinkerPop compatible databases as well.

File: proxy-server.js [hunk 1]
```diff
@@ -2,66 +2,127 @@ const express = require('express');
...
+                logger.debug(result);
+                res.send(nodesToJson(result._items))
+            })
+            .catch((err) => {
+                logger.warn(err);
+                next(err)
+            });
+    } else {
+        const safeQuery = querystring.escape(realQuery);
+        const url = `http://${gremlinHost}:${gremlinPort}?gremlin=${safeQuery}`;
```
Comment (triggan) [inline proxy-server.js:107]: The url here needs to be http(s)://${gremlinHost}:${gremlinPort}/gremlin?gremlin=${safeQuery}

The endpoint for most TinkerPop databases is /gremlin in addition to a gremlin parameter which is passed with the query to that endpoint.   Not sure how best we want to handle http/https.  Potentially make that another configurable?

Comment (thiagomata) [reply proxy-server.js]: Fixed, please take a look.

**Pull request closed without merging**